# Q2(b) - Belt

In [39]:
from math import pi, sin, cos, sqrt, pow, radians, degrees, asin, e
import numpy as np

In [40]:
Length=4.27
Power =7500
n1=1000
n2=500
centerDistance = 1.5
friction=0.25
print(f'From diagram, center distance is {centerDistance}m')

From diagram, center distance is 1.5m


i) Determine pulley diameter and cross-sectional area of belt K

Use belt length equation

$$L=2c+\pi(r_1+r_2)+\frac{1}{c}(r_2-r_1)^2$$
$$4.27=2(1.5)+\pi(r_1+r_2)+\frac{1}{1.5}(r_2-r_1)^2$$

Find r1 equation

$$\frac{n_1}{n_2}=\frac{1000}{500}=\frac{r_2}{r_1}$$

From this equation

$$r_2=2r_1$$

Substitute into belt length equation


$$4.27=2(1.5)+\pi(r_1+2r_1)+\frac{1}{1.5}(2r_1-r_1)^2$$
$$4.27=3+\pi(3r_1)+\frac{1}{1.5}(r_1)^2$$


$$0.67r_1^2+9.42r_1-1.27$$

Solve quadratic equation using

$$ax^2+bx+c=0 \rightarrow x=\frac{-b\pm \sqrt{b^2-4c}}{2a}$$

In [41]:
def solve_quadratic_numpy(a, b, c):
  """Solves a quadratic equation using numpy.roots."""
  coefficients = [a, b, c]  # Coefficients in descending order of power
  roots = np.roots(coefficients)
  return roots[0], roots[1]  # Return the two roots

a = 0.67
b = 9.42
c = -1.27

x1, x2 = solve_quadratic_numpy(a, b, c)
print(f"The roots are: x1 = {x1}, x2 = {x2}")
r1=round(x2,3)
print(f'The possible value for r1 is {r1} m')
r2=2*r1
print(f'r2={r2} m')

The roots are: x1 = -14.19325244538296, x2 = 0.13355095284564683
The possible value for r1 is 0.134 m
r2=0.268 m


In [42]:
d1=2*r1
d2=2*r2
print(f'Diameter for motor = {d1} m')
print(f'Diameter for Pulley A = {d2} m')

Diameter for motor = 0.268 m
Diameter for Pulley A = 0.536 m


$$sin\alpha = \frac{r_2-r_1}{c}$$
$$sin\alpha = \frac{0.268-0.134}{1.5}$$

In [43]:
alpha = asin((r2-r1)/centerDistance)
print(f'alpha={alpha}')
print(f'alpha in degrees = {degrees(alpha)}')
smallPulley = round(pi-(2*alpha),3)
largePulley = round(pi+(2*alpha),3)
print(f'Small pulley, phi={smallPulley} rad')
print(f'Small pulley, phi={degrees(smallPulley)}')
print(f'Large pulley, phi={largePulley} rad')
print(f'Large pulley, phi={degrees(largePulley)}')

alpha=0.08945258202819847
alpha in degrees = 5.12525541676357
Small pulley, phi=2.963 rad
Small pulley, phi=169.76739469726292
Large pulley, phi=3.32 rad
Large pulley, phi=190.2219879834333


Use Power equation
$$P=\omega T$$
$$P=\frac {2\pi n_1}{60}T$$

$n_1=1000rpm$

$$7500=\frac {2\pi n_1}{60}T$$

In [44]:
Torque=round(Power*60/(2*pi*n1),2)
print(f'Torque={Torque} Nm')

Torque=71.62 Nm


From $F_1$ equation
$$F_1=F_c+\left[ \frac{\gamma}{\gamma-1} \right]\frac{T_1}{r_1}$$
where
$$\gamma =e^\frac{f \phi}{sin \beta}$$
For flat belt, $\beta=\frac{180}{2}=90$

Need to find $F_c$ first
$$F_c = \rho A V^2$$

and

$$V = \frac {\pi dn_{rpm}}{60}$$

In [47]:
velocity = round(pi*d1*n1/60,2)
print(f'Velocity, V={velocity}m/s')

Velocity, V=14.03m/s


In [53]:
beltDensity=950
Fc = round(beltDensity*pow(velocity,2),1)
print(f'Fc ={Fc}A')
print('Left Fc with unknown A')

Fc =186998.9A
Left Fc with unknown A


Related equations to cross-sectional area
$$\sigma = \frac{F_{max}}{A}$$
$$\sigma = \frac{K_s F_1}{A}$$

No service factor. Therefor Ks=1

$$1.75MPa = \frac{1\times F_1}{A}$$

In [49]:
print(f'F1={1.75E+6}A')

F1=1750000.0A


Substitute equation

$$F_1=1750000A$$

into

$$F_1=F_c+\left[ \frac{\gamma}{\gamma-1} \right]\frac{T_1}{r_1}$$

Find gamma first


In [50]:
print(f'phi = {smallPulley}')
gamma = round(pow(e,friction*smallPulley/sin(radians(90))),4)
print(f'gamma={gamma}')

phi = 2.963
gamma=2.0975


Substitute $\gamma$ to F1 equation
$$F_1=186998.9A+\left[ \frac{2.0975}{2.0975-1} \right]\frac{71.62}{0.134}$$
$$1750000A=186998.9A+\left[ \frac{2.0975}{2.0975-1} \right]\frac{71.62}{0.134}$$

In [54]:
rightSide = (gamma/(gamma-1))*Torque/r1

print(f'{1.75E+6}A = {Fc}A+{round(rightSide,2)}')

1750000.0A = 186998.9A+1021.47


The equation become

$$1750000.0A = 186998.9A+1021.47$$

In [56]:
Area = rightSide/(1.75E+6-Fc)
print(f'Cross-sectional area of belt, A={Area:.2e}m^2')

Cross-sectional area of belt, A=6.54e-04m^2
